cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/SparkRuntimeService.java

```diff
@@ -167,11 +167,7 @@ protected void run() throws Exception {
         throw e;
       }
     } finally {
-      try {
-        Closeables.closeQuietly(executionContext);
-      } finally {
-        Thread.currentThread().setContextClassLoader(oldClassLoader);
-      }
```

Comment:
Removed this because the transaction gets committed in shutdown(). If the datasets are already closed at that time, they will throw an exception in commitTx(), and the whole job fails.